source/vfs/vfs.go

```diff
@@ -0,0 +1,141 @@
+// Package vfs contains a driver that reads migrations from a virtual file
+// system.
+//
+// Implementations of the filesystem interface that read from zip files and
+// maps, as well as the definition of the filesystem interface can be found in
+// the golang.org/x/tools/godoc/vfs package.
+package vfs
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/golang-migrate/migrate/source"
+	"golang.org/x/tools/godoc/vfs"
+)
+
+func init() {
+	source.Register("vfs", &VFS{})
```

Comment:
Do you know if the `vfs` name will conflict with other virtual filesystem names?

This is not an issue with a mounted virtual filesystem since it'll appear as any other file/path but access to an unmounted virtual filesystems will require a new source driver. I don't know of any implementations that use the `vfs` acronym.

We could be ultra conservative and use `go-vfs` as the driver name (you'll need to use `go_vfs` as the build constraint, package name, and file name(s))

Reply:
I think `go-vfs` is less likely to have conflicts than `vfs` so let's use that name.
e.g. Use `go-vfs` as the registered driver name and `go_vfs` as the build constraint name, source driver package name, and filenames.

Reply:
Does the package actually need to be named that? It feels silly to have Go in the package name; what other language would it be?

Reply:
Unfortunately, as it stands today, the tests need to use the same package names as the build constraint names.
See `test-with-flags` in the Makefile.

We could update the Makefile to test packages separately from build constraint names, but I like the DRY principle. Eventually, I want to run tests using `go test ./...` instead of explicitly listing out tests to run.

I'm fine w/ the package name `godoc_vfs`. Don't forget to use `godoc-vfs` as the driver name, which means sources will be specified as `godoc-vfs://`

Reply:
Done